### PR TITLE
Add entry points to the registration view

### PR DESCRIPTION
**Pull request by ghickman**

### CODE REVIEW

**Comment by ghickman** [comment]
@incuna/backend Could you take a look at this please.

It's related to https://github.com/incuna/roche-oncology/pull/420 which is currently setting `user.is_active` to False in the post registration signal and adding a second db hit on the user.

File: profiles/views.py [hunk 1]
```diff
@@ -73,13 +82,13 @@ def form_valid(self, form):
...
 
-        if password:
-            user = authenticate(username=form.cleaned_data['username'], password=password)
-            messages.info(self.request, _('Your profile has been created.'))
-            login(self.request, user)
+        self.authenticate_new_user(self, form.cleaned_data['username'], password)
 
         return super(RegisterView, self).form_valid(form)
 
+    def update_user_object(self, user):
```
Comment (maxpeterson) [inline profiles/views.py:92]: As you mentioned this should return obj.

File: profiles/views.py [hunk 2]
```diff
@@ -35,6 +35,13 @@ class RegisterView(ProfleFormMixin, CreateView):
     template_name = 'profiles/profile_form.html'
     success_url = getattr(settings, 'REGISTRATION_COMPLETE_URL', settings.LOGIN_REDIRECT_URL)
 
+    def authenticate_new_user(self, user, password):
```
Comment (maxpeterson) [inline profiles/views.py:38]: Maybe this should return the user

**Comment by ghickman** [comment]
@maxpeterson updated as suggested and pulled username/password out too.

File: profiles/views.py [hunk 3]
```diff
@@ -35,6 +35,13 @@ class RegisterView(ProfleFormMixin, CreateView):
...
+        if password:
+            user = authenticate(username=user.username, password=password)
+            messages.info(self.request, _('Your profile has been created.'))
+            login(self.request, user)
+            return user
```
Comment (maxpeterson) [inline profiles/views.py]: Shouldn't it return the user `if not password`

**Pull request merged**